ui/src/main/webapp/home/reducer.js

```diff
@@ -0,0 +1,30 @@
+import { combineReducers } from 'redux-immutable'
+import { Map } from 'immutable'
+
+export const getSourceConfigs = (state) => state.getIn(['home', 'configs', 'sources'])
+export const getLdapConfigs = (state) => state.getIn(['home', 'configs', 'ldap'])
+export const getSubmittingPids = (state) => state.getIn(['home', 'submitting']).toJS()
+
+const configs = (state = Map(), { type, id, servicePid, value = [] }) => {
+  switch (type) {
+    case 'HOME/SET_CONFIGS':
+      return state.set(id, value)
+    default:
+      return state
+  }
+}
+
+const submitting = (state = Map(), { type, servicePid, value = [] }) => {
+  switch (type) {
+    case 'HOME/BEGIN_SUBMITTING_CONFIGS':
+      return state.set(servicePid, true)
+    case 'HOME/END_SUBMITTING_CONFIGS':
```

Comment:
No, that case is not being used right now. I kept it in here in case there were any other actions that needed to turn off the spinner without having to update the configs at the same time. When I add the error message display to the home page you mentioned before, then this would be used.